frontend/tickers.go

```diff
@@ -0,0 +1,46 @@
+package main
+
+import (
+	"context"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/grpcoin/grpcoin/gdax"
+)
+
+type tickerService struct {}
+
+var newTicker tickerService
+
+func wsTickers(w http.ResponseWriter, r *http.Request) {
+	conn, err := upgrader.Upgrade(w, r, nil)
+	defer conn.Close()
+	if err != nil {
+		log.Println(err)
+		return
+	}
+
+	client := &Client{hub: hub, conn: conn, send: make(chan []byte, 256)}
+	client.hub.register <- client
+
+	ctx, stop := context.WithTimeout(context.Background(), time.Hour*5)
+
+	quotes, err := gdax.StartWatch(ctx, "BTC-USD", "ETH-USD")
+	if err != nil {
+		stop()
+		return
+	}
```

Comment:
for each client this causes a new and separate connection to coinbase.

to do this efficiently we need to extract code from tickerservece.go#Watch (initWatch ...).

Reply:
extract pubsub.go pubsub pkg